### PR TITLE
NASR cross-check warnings for airports.json elevation and magnetic declination

### PR DESCRIPTION
**Pull request by alexwitherspoon**
## Summary

Offline maintainer tooling to catch stale `elevation_ft` and per-airport `magnetic_declination` in airports.json before it affects display or DA math. Closes #200.

- **`lib/nasr/parse.php`** - parse `MAG_VARN` / `MAG_HEMIS` / `MAG_VARN_YEAR` into signed `mag_declination_deg` (East positive)
- **`lib/nasr/config-cross-check.php`** - warning-only compare against NASR when a row exists (±2 ft elevation, ±1.0° magnetic); skips global WMM and missing fields
- **`scripts/config-check.php`** - replaces inline Makefile PHP; supports `CONFIG_PATH`; exit 0 on warnings
- **`scripts/validate-airports-json.php`** - optional step 4 when NASR cache is present (secrets CI stays green when cache is absent)
- Tolerances in `lib/constants.php`; docs in `docs/OPERATIONS.md`

No auto-fix, no API/runtime mutation, no deploy blockers.

## Test plan

- [x] `make test-ci` passes locally
- [x] `NasrParseTest` and `NasrConfigCrossCheckTest` (match, mismatch, skip paths)
- [x] `php scripts/config-check.php` with NASR cache present (warnings only, exit 0)
- [x] `CONFIG_PATH=../aviationwx.org-secrets/airports.json make config-check` - 51 NASR rows checked, 0 warnings on current fleet
- [x] Docker test container: `config-check` and validate step 4; graceful skip when NASR cache removed

### CODE REVIEW

File: lib/nasr/config-cross-check.php [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+    if (!nasrCrossCheckHasMagneticComparison($airport, $nasrRecord)) {
+        return null;
+    }
+
+    $configDeg = (float) $airport['magnetic_declination'];
+    $nasrDeg = (float) $nasrRecord['mag_declination_deg'];
+    $delta = abs($configDeg - $nasrDeg);
+    $tolerance = NASR_CONFIG_MAGNETIC_TOLERANCE_DEG;
+
+    if ($delta < $tolerance) {
```
[the remark this answers is not in the document]
Comment (alexwitherspoon) [reply lib/nasr/config-cross-check.php]: Good catch — agreed. Magnetic now uses `<=` like elevation in 6e660c0b, so exactly 1.0° is within tolerance and we only warn when delta exceeds ±1.0°.

File: tests/Unit/NasrConfigCrossCheckTest.php [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+                    'magnetic_declination' => 15,
+                ],
+            ],
+        ]);
+
+        $this->assertCount(1, $result['warnings']);
+        $this->assertStringContainsString("Airport 'keul' (EUL)", $result['warnings'][0]);
+        $this->assertStringContainsString('magnetic_declination 15 differs from NASR 16°E (year 2000)', $result['warnings'][0]);
+        $this->assertSame(1, $result['summary']['magnetic_warnings']);
+    }
```
[the remark this answers is not in the document]
Comment (alexwitherspoon) [reply tests/Unit/NasrConfigCrossCheckTest.php:91]: Updated in 8a3f9178: mismatch test uses 14° vs NASR 16°E (2° delta), and I added an explicit boundary case at 15° vs 16°E that should pass without warning.

File: scripts/validate-airports-json.php [hunk 3]
```diff
@@ -119,6 +120,25 @@
     echo "  ✓ All ICAO codes are valid\n";
 }
 
+// Validation 4: NASR cross-check (warnings only, non-blocking)
+echo "\n4. NASR cross-check (elevation_ft, magnetic_declination)...\n";
+if (loadNasrAptCache() === null) {
+    echo "  NASR cache not present; skipping elevation/magnetic cross-check\n";
+} else {
+    $nasrResult = nasrCrossCheckAirportConfig($config);
```
[the remark this answers is not in the document]
Comment (alexwitherspoon) [reply scripts/validate-airports-json.php:135]: Valid point — fixed in a4527f73. Step 4 now sets CONFIG_PATH to the resolved CLI airports.json path and resets the NASR memo before loadNasrAptCache(), so the configured slice matches the file being validated.

File: tests/Unit/NasrConfigCrossCheckTest.php [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+    {
+        $this->loadNasrAptFixtureCache();
+    }
+
+    protected function tearDown(): void
+    {
+        $this->tearDownNasrAptFixtureCache();
+    }
+
+    public function testElevationMatchWithinToleranceProducesNoWarning(): void
```
[the remark this answers is not in the document]
Comment (alexwitherspoon) [reply tests/Unit/NasrConfigCrossCheckTest.php:25]: I am keeping camelCase here to match the other NASR unit tests in this area (NasrParseTest, NasrConfiguredCacheTest, NasrCacheHealthTest). The underscore pattern in CODE_STYLE is what we use for safety-critical tests like MagneticDeclinationSafetyTest; renaming only this file would make the NASR suite inconsistent.
